bot/code_review_bot/workflow.py

```diff
@@ -288,6 +322,41 @@ def index(self, revision, **kwargs):
                 },
             )
 
+    def find_previous_issues(self, issues, base_rev_changeset=None):
+        """
+        Look for known issues in the backend matching the given list of issues
+
+        If a base revision ID is provided, compare to issues detected on this revision
+        Otherwise, compare to issues detected on last ingested revision
+        """
+        assert (
+            self.backend_api.enabled
+        ), "Backend storage is disabled, comparing issues is not possible"
+
+        current_date = datetime.now().strftime("%Y-%m-%d")
+
+        # Group issues by path, so we only list know issues for the affected files
+        groups = defaultdict(list)
+        for i in issues:
+            groups[i.path].append(i)
```

Comment:
Just use itertools.groupby

Reply:
I'm not sure paths are consecutive here (e.g. it would create multiple groups if different analyzers detected issues on the same file). I noticed groupby is used somewhere else to group by analyzer.

Reply:
Can you sort by path here first, then group by ?